### PR TITLE
Expose cass-operator AdditionalServiceConfig to k8ssandra helm chart

### LINKED ISSUE #874: Allow for custom labels and annotations in values.yaml
Allow for custom `labels` and `annotations` for the `Cassandra` services. For example `Service Mesh` like `Istio`, `Consul` and others require custom annotations to work.

┆Issue is synchronized with this [Jiraserver Feature](https://k8ssandra.atlassian.net/browse/K8SSAND-567) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-567
┆Priority: Medium

### PR DESCRIPTION
**Pull request by gerbal**
**What this PR does**:

This PR exposes the Cass-Operator variable [AdditionalServiceConfig](https://github.com/k8ssandra/cass-operator/blob/148b57fab4bd25920cd4400895cb494216f84cf0/charts/cass-operator-chart/templates/customresourcedefinition.yaml#L42) to the k8ssandra helm chart.

**Which issue(s) this PR fixes**:
Fixes #874

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by jsanda — CHANGES_REQUESTED** [review]
Hi @gerbal 

Thanks for the PR! I have a couple asks. 

1) Can you update the 1.4 CHANGELOG? 

2) Can you add some unit tests?

File: tests/unit/template_cassdc_test.go [hunk 1]
```diff
@@ -551,6 +551,16 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
...
 		})
 
+		It("Setting additionaServiceConfig", func() {
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				ValuesFiles:    []string{"./testdata/additional-service-config-values.yaml"},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			Expect(cassdc.Spec.AdditionalServiceConfig).ToNot(BeNil())
```
Comment (jsanda) [inline tests/unit/template_cassdc_test.go:561]: Thanks for adding the test. Can you add verification that `AdditionalServiceConfig` matches the expected values? I realize that could get a bit tedious but doing so has a couple benefits. First, it validates and demonstrates how the chart property should be used. Secondly it make it safe to refactor that part of the template in the future should the need arise.

**Review by jsanda — CHANGES_REQUESTED** [review]

File: tests/unit/template_cassdc_test.go (hunk 1 above)
Comment (gerbal) [reply tests/unit/template_cassdc_test.go:561]: New assertions added to verify additional service config values

**Review by jsanda — APPROVED** [review]
@gerbal nice work!

**Comment by jdonenine** [comment]
Thanks for the contribution @gerbal ! 🥳

**Pull request merged**